### ISSUE TITLE
Integrate with https://fhirpath-lab.com/FhirPath

### ISSUE THREAD

**Issue by ir4y** [beda-software/fhirpath-py#30]
https://github.com/brianpos/fhirpath-lab/blob/develop/server-api.md

**Comment by ir4y** [comment]
Some comments from **Brian Postlethwaite**

These are the logical models for what data is in/out of the custom operation
Simplified logical models for what data in the request/response.  

<img width="276" alt="image(1)" src="https://github.com/beda-software/fhirpath-py/assets/1488983/90bd5fe3-9ea4-4a83-afc4-315dbf0e4b67">   

  
<img width="398" alt="image(2)" src="https://github.com/beda-software/fhirpath-py/assets/1488983/8f2815cb-7bc1-47ce-995a-e5dfd111b9c3">   


But represented using fhir parameters resources.
I'm trying to get real profiles for these to go with things too.
The Abstract Syntax tree format isn't documented anywhere. I built it from the Firely engine, and mapped the fhirpath.js to that. In progress mapping the HAPI engine to it too.
The tricky part is making it work with the "invert tree" format to make it more readable.

**Comment by ir4y** [comment]
From https://github.com/brianpos/

Hey Ilya, I tried out the branch with the python engine in it and looks like it works reasonable well.
I did note that your results are coming out as a single result.
If a context expression comes through, it should return multiple results, one for each result from the context expression.
Also, you might like to append the version number of your engine in the string that you return for the evaluator (if you have one)
e.g. `"valueString": "fhirpath-py"` => `"valueString": "fhirpath-py (x.y.z)"`
This is from the HAPI engine
[image.png](https://chat.fhir.org/user_uploads/10155/XwAo79XTbUM-R2ruCRju6jMx/image.png)
This is from yours
[image.png](https://chat.fhir.org/user_uploads/10155/qGVlVi1Xftxg-75X1kozPQdi/image.png)
